packages/apps-sdk/src/sdk.ts

```diff
@@ -31,9 +34,19 @@ export class SDK {
    */
   private configureApp(): void {
     // First we obtaing the id of this app from the url
-    const parsed = queryString.parse(window.location.search);
+    const parsed = qs.parse(window.location.search.substring(1), { decoder });
     const parsedAppName = Array.isArray(parsed.appName) ? parsed.appName[0] : parsed.appName;
+
+    this.queryParams = Array.isArray(parsed.queryParams) ? parsed.queryParams[0] : parsed.queryParams;
+    this.initialData = Array.isArray(parsed.initialData) ? parsed.initialData[0] : parsed.initialData;
     this.appName = parsedAppName ? parsedAppName : 'messenger-iframe';
+
+    const mode = this.queryParams && this.queryParams.mode ? this.queryParams.mode : APP_MODE_IFRAME;
+    const channelConfig = mode === APP_MODE_POPUP ? { window: window.opener } : { window: window.parent };
```

Comment:
what if I write down `mode: puppy` by mistake in the `queryParams.mode`? it won't have any `channelConfig`?

Reply:
It would use default IFRAME